launcher/cog-launcher.c

```diff
@@ -238,11 +238,14 @@ cog_launcher_create_view(CogLauncher *self, CogShell *shell)
     }
 
     g_autoptr(GError) error = NULL;
-    CogPlatform      *platform =
-        cog_platform_configure(s_options.platform_name, s_options.platform_params, "COG", shell, &error);
+    CogPlatform      *platform = cog_platform_new(s_options.platform_name ?: g_getenv("COG_PLATFORM_NAME"), &error);
```

Comment:
cog_platform_new doesn't check if there is already created a platform.

Reply:
The order doesn't really matter anymore: creating a view goes through `cog_view_get_impl_type()`, which uses `cog_platform_get()`, and at that point the platform singleton will be created. The only situation in which the platform singleton needs to be created first is to manually choosing the platform plug-in *using the API*&mdash;environment variables are still honored when letting `cog_platform_get()` create the singleton.

TL;DR with the new approach the only ordering restriction is “call `cog_init()` before anything else if you want to choose the platform plug-in using the API”.